Sources/Client/Client/ClientLogger.swift

```diff
@@ -227,6 +229,32 @@ public final class ClientLogger {
         lastTime = startTime
     }
     
+    public func logTaskDuration(_ description: String, task: () -> Void) {
+        let id = logTaskStarted(description)
+        task()
+        logTaskFinished(taskId: id)
+    }
+    
+    @discardableResult
+    public func logTaskStarted(_ description: String, taskId: UUID = .init()) -> UUID {
+        measuredTasksStart.update {
+            var tasks = $0
+            tasks[taskId] = (start: Date(), description: description)
+            return tasks
+        }
+        return taskId
+    }
+
+    public func logTaskFinished(taskId id: UUID) {
+        guard let task = measuredTasksStart[id] else {
+            log("Trying to finish a logger measuring task with \(id) but the task wasn't started.", level: .error)
+            return
+        }
+        
+        let duration = Date().timeIntervalSince1970 - task.start.timeIntervalSince1970
+        log("⏱ \(task.description) finished in \(duration) seconds.", level: .debug)
```

Comment:
Should we introduce a special option for this? Something like `.performance`?

Reply:
I don't think we should do that. It's not so much noise and actually in early logger version v1 had measurements, but didn't have elegant implementation like this.
I think developers could check JSON output to understand numbers.